helix-core/src/line_ending.rs

```diff
@@ -13,6 +13,29 @@ pub enum LineEnding {
     PS,   // U+2029 -- ParagraphSeparator
 }
 
+impl LineEnding {
+    pub fn len(&self) -> usize {
+        match self {
+            Self::Crlf => 2,
+            _ => 1,
+        }
+    }
+
+    pub fn as_str(&self) -> &str {
+        match self {
+            Self::Crlf => "\u{000D}\u{000A}",
+            Self::LF => "\u{000A}",
+            Self::Nel => "\u{0085}",
+            Self::LS => "\u{2028}",
+            Self::CR => "\u{000D}",
+            _ => panic!(
+                "Unexpected line ending: {:?}, expected Crlf, LF, CR, Nel, or LS.",
+                self
+            ),
+        }
+    }
```

Comment:
Is there a reason for this to not handle all of the variants?  This seems like a generally useful function, and I wouldn't expect it to panic just because those variants aren't output by the line ending detector.

Reply:
Whoops, sorry for the double-comment.  The other one wasn't showing up in the diff view for some reason.

Reply:
Uhh, no idea why I excluded some. Something to do with only matching standard line endings in some previous iteration of the PR. You are right, we should be just handling everything here.